src/access/universal.ts

```diff
@@ -32,3 +61,317 @@ export interface Access {
+ *   resolve to `null`.
+ * - It will only return access specified explicitly for the given Agent. If
+ *   additional restrictions are set up to apply to the given Agent in a
+ *   particular situation, those will not be reflected in the return value of
+ *   this function.
+ * - It will only return access specified explicitly for the given Resource.
+ *   In other words, if the Resource is a Container, the returned Access may not
+ *   apply to contained Resources.
+ * - If the current user does not have permission to view access for the given
+ *   Resource, this function will resolve to `null`.
+ *
+ * @param resourceUrl URL of the Resource you want to read the access for.
+ * @param webId WebID of the Agent you want to get the access for.
+ */
+export async function getAgentAccess(
+  resourceUrl: UrlString,
+  webId: WebId,
+  options = internal_defaultFetchOptions
+): Promise<Access | null> {
+  const resourceInfo = await getResourceInfoWithAcr(resourceUrl, options);
+  if (hasAccessibleAcr(resourceInfo)) {
+    return getAgentAccessAcp(resourceInfo, webId);
+  }
+  if (hasAccessibleAcl(resourceInfo)) {
+    return await getAgentAccessWac(resourceInfo, webId, options);
+  }
+  return null;
+}
+
+/**
+ * Set access to a Resource for a specific Agent.
+ *
+ * This function works with Solid Pods that implement either the Web Access
+ * Control spec or the Access Control Policies proposal, with some caveats:
+ *
+ * - If access to the given Resource has been set using anything other than the
+ *   functions in this module, it is possible that it has been set in a way that
+ *   prevents this function from reliably setting access, in which case it will
+ *   resolve to `null`.
+ * - It will only set access explicitly for the given Agent. In other words,
+ *   additional restrictions could be present that further restrict or loosen
+ *   what access the given Agent has in particular circumstances.
+ * - The provided access will only apply to the given Resource. In other words,
+ *   if the Resource is a Container, the configured Access may not apply to
+ *   contained Resources.
+ * - If the current user does not have permission to view or change access for
+ *   the given Resource, this function will resolve to `null`.
+ *
+ * Additionally, two caveats apply to users with a Pod server that uses WAC:
+ * - If the Resource did not have an ACL yet, a new one will be initialised.
+ *   This means that changes to the ACL of a parent Container can no longer
+ *   affect access people have to this Resource, although existing access will
+ *   be preserved.
+ * - Setting different values for `controlRead` and `controlWrite` is not
+ *   supported, and **will throw an error**. If you expect (some of) your users
+ *   to have Pods implementing WAC, be sure to pass the same value for both.
+ * - Denying access is not technically possible in WAC. Rather, what this does
+ *   is removing "allow" access.
```

Comment:
When we do the same for ACP (i.e. `unallow` instead of `deny`), is this description still going to be relevant ? If not, I think I'd add a note to the related ticket to update it. This remark applies to other comments in this file as well.

Reply:
Yeah good point, I suppose it's technically most accurate to update that together with that ticket. I'll add a note.